src/_langs/en/updates/2015-06-12-Media-Source-Extensions-for-Audio.markdown

```diff
@@ -0,0 +1,373 @@
+
+We'll get into the details of why below, but for now lets start with a demonstration. Below is the first thirty seconds of the excellent [Sintel](http://www.sintel.org/) chopped into five separate MP3 files and reassembled using MSE. The red lines indicate gaps introduced during the creation (encoding) of each MP3; you'll hear glitches at these points.
+
+<iframe seamless style="border: 0; height: 335px; width: 100%" src="https://simpl.info/mse/audio/gap"></iframe>
+
+Yuck! That's not a great experience; we can do better. With a little more work, using the exact same MP3 files in the above demo, we can use MSE to remove those annoying gaps. The green lines in the next demo indicate where the files have been joined and the gaps removed. On Chrome 38+ this will playback seamlessly!
+
+<iframe seamless style="border: 0; height: 335px; width: 100%" src="https://simpl.info/mse/audio/gapless"></iframe>
+
+There are a [variety of ways to create gapless content](#creating-gapless-content). For the purposes of this demo, we'll focus on the type of files a normal user might have lying around. Where each file has been encoded separately without regard for the audio segments before or after it.
+
+## Basic Setup
+
+First, lets backtrack and cover the basic setup of a MediaSource instance. Media Source Extensions, as the name implies, are just extensions to the existing media elements. Below, we're assigning an [Object URL](https://developer.mozilla.org/en-US/docs/Web/API/URL.createObjectURL), representing our MediaSource instance, to the source attribute of an audio element; just like you would set a standard URL.
+
+{% highlight javascript %}
+var audio = document.createElement('audio');
+var mediaSource = new MediaSource();
+var SEGMENTS = 5;
+
+mediaSource.addEventListener('sourceopen', function() {
+  var sourceBuffer = mediaSource.addSourceBuffer('audio/mpeg');
+
+  function onAudioLoaded(data, index) {
+    // Append the ArrayBuffer data into our new SourceBuffer.
+    sourceBuffer.appendBuffer(data);
+  }
+
+  // Retrieve an audio segment via XHR.  For simplicity, we're retrieving the
+  // entire segment at once, but we could also retrieve it in chunks and append
+  // each chunk separately.  MSE will take care of assembling the pieces.
+  GET('sintel/sintel_0.mp3', function(data) { onAudioLoaded(data, 0); } );
+}, false);
+
+audio.src = window.URL.createObjectURL(mediaSource);
+{% endhighlight %}
+
+Once the MediaSource object is connected, it will perform some initialization and eventually fire a "sourceopen" event; at which point we can create a [SourceBuffer](http://www.w3.org/TR/media-source/#sourcebuffer). In the example above, we're creating an "audio/mpeg" one, which is able to parse and decode our MP3 segments; there are several [other types](http://www.w3.org/2013/12/byte-stream-format-registry/) available.
+
```

Comment:
Good spot! Found a few of these. Changed.

On 19 June 2015 at 18:24, Joe Medley <notifications@github.com> wrote:

> In
> src/_langs/en/updates/2015-06-12-Media-Source-Extensions-for-Audio.markdown
> <https://github.com/google/WebFundamentals/pull/1381#discussion_r32849947>
> :
>
> > +  function onAudioLoaded(data, index) {
> > +    // Append the ArrayBuffer data into our new SourceBuffer.
> > +    sourceBuffer.appendBuffer(data);
> > +  }
> > +
> > +  // Retrieve an audio segment via XHR.  For simplicity, we're retrieving the
> > +  // entire segment at once, but we could also retrieve it in chunks and append
> > +  // each chunk separately.  MSE will take care of assembling the pieces.
> > +  GET('sintel/sintel_0.mp3', function(data) { onAudioLoaded(data, 0); } );
> > +}, false);
> > +
> > +audio.src = window.URL.createObjectURL(mediaSource);
> > +{% endhighlight %}
> > +
> > +Once the MediaSource object is connected, it will perform some initialization and eventually fire a "sourceopen" event; at which point we can create a [SourceBuffer](http://www.w3.org/TR/media-source/#sourcebuffer). In the example above, we're creating an "audio/mpeg" one, which is able to parse and decode our MP3 segments; there are several [other types](http://www.w3.org/2013/12/byte-stream-format-registry/) available.
> > +
>
> Please replace quotes on sourceopen with backticks(`) to give it a
> different type face.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/WebFundamentals/pull/1381/files#r32849947>.
>